pennylane/utils.py

```diff
@@ -150,12 +150,12 @@ def sparse_hamiltonian(H, wires=None):
         wires = qml.wires.Wires(wires)
 
     n = len(wires)
-    matrix = scipy.sparse.coo_matrix((2**n, 2**n), dtype="complex128")
+    matrix = scipy.sparse.csr_matrix((2**n, 2**n), dtype="complex128")
```

Comment:
I like that, with this, we can have a CSR matrix from the beginning because further away you'll be summing temporary (denser) matrices already in the CSR format. This makes good use of the nice performance the CSR format brings to Arithmetic.